Content.Client/Administration/UI/Bwoink/BwoinkPanel.xaml.cs

```diff
@@ -53,5 +63,54 @@ public void ReceiveLine(SharedBwoinkSystem.BwoinkTextMessage message)
             TextOutput.AddMessage(formatted);
             LastMessage = message.SentAt;
         }
+
+        private void UpdateTypingIndicator()
+        {
+            var msg = new FormattedMessage();
+            msg.PushColor(Color.LightGray);
+
+            var text = PeopleTyping.Count == 0
+                ? string.Empty
+                : Loc.GetString("bwoink-system-typing-indicator",
+                    ("players", string.Join(", ", PeopleTyping)),
+                    ("count", PeopleTyping.Count));
+
+            msg.AddText(text);
+            msg.Pop();
+
+            TypingIndicator.SetMessage(msg);
+        }
+
+        public void UpdatePlayerTyping(string name, bool typing)
+        {
+            if (typing)
+            {
+                if (PeopleTyping.Contains(name))
+                    return;
+
+                PeopleTyping.Add(name);
+                Timer.Spawn(TimeSpan.FromSeconds(15), () =>
```

Comment:
15 seconds seems too much to me, perhaps making it 10 like in discord?

Reply:
Fixed in [`c1e27a3` (#19060)](https://github.com/space-wizards/space-station-14/pull/19060/commits/c1e27a3c7b740d60b9df0b747528a14f94627f48)